renderdoc/driver/metal/metal_types.h

```diff
@@ -434,6 +476,55 @@ struct RenderPassDescriptor
   rdcarray<RenderPassSampleBufferAttachmentDescriptor> sampleBufferAttachments;
 };
 
+// MTLComputePassSampleBufferAttachmentDescriptor : based on the interface defined in
+// Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/System/Library/Frameworks/Metal.framework/Headers/MTLComputePass.h
+struct ComputePassSampleBufferAttachmentDescriptor
+{
+  ComputePassSampleBufferAttachmentDescriptor() = default;
+  ComputePassSampleBufferAttachmentDescriptor(MTL::ComputePassSampleBufferAttachmentDescriptor *objc);
+  void CopyTo(MTL::ComputePassSampleBufferAttachmentDescriptor *objc);
+  // TODO: when WrappedMTLCounterSampleBuffer exists
+  // MTLCounterSampleBuffer *sampleBuffer = NULL;
+  NS::UInteger startOfEncoderSampleIndex = MTLCounterDontSample;
+  NS::UInteger endOfEncoderSampleIndex = MTLCounterDontSample;
+};
+
+// MTLComputePipelineDescriptor : based on the interface defined in
+// Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/System/Library/Frameworks/Metal.framework/Headers/MTLComputePipeline.h
+struct ComputePipelineDescriptor
+{
+  ComputePipelineDescriptor() = default;
+  ComputePipelineDescriptor(MTL::ComputePipelineDescriptor *objc);
+  explicit operator MTL::ComputePipelineDescriptor *();
+  rdcstr label;
+  WrappedMTLFunction *computeFunction = NULL;
+  bool threadGroupSizeIsMultipleOfThreadExecution = false;
+  NS::UInteger maxTotalThreadsPerThreadgroup = 1;
+  NS::UInteger maxCallStackDepth = 1;
+  StageInputOutputDescriptor stageInputDescriptor;
+  rdcarray<RDMTL::PipelineBufferDescriptor> buffers;
+  bool supportIndirectCommandBuffers = false;
+  // TODO: when WrappedMTLDynamicLibrary exists
+  // rdcarray<WrappedMTLDynamicLibrary*> preloadedLibraries;
+  // Deprecated
+  // rdcarray<WrappedMTLDynamicLibrary*> insertLibraries;
+  RDMTL::LinkedFunctions linkedFunctions;
+  bool supportAddingBinaryFunctions = false;
+  // TODO: when WrappedMTLBinaryArchive exists
+  // rdcarray<WrappedMTLBinaryArchive*> binaryArchives;
+};
+
+// MTLComputePassDescriptor : based on the interface defined in
+// Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/System/Library/Frameworks/Metal.framework/Headers/MTLComputePass.h
+struct ComputePassDescriptor
+{
+  ComputePassDescriptor() = default;
+  ComputePassDescriptor(MTL::ComputePassDescriptor *objc);
+  explicit operator MTL::ComputePassDescriptor *();
+  rdcarray<ComputePassSampleBufferAttachmentDescriptor> sampleBufferAttachments;
+  MTL::DispatchType dispatchType = MTL::DispatchTypeSerial;
```

Comment:
No need to change just explaining a development principle I have been using. 

The documentation isn't fully clear and I don't trust documentation to match code behaviour as a rule.
In the past I have called the ObjC create methods i.e. `computePassDescriptor` to see what defaults it makes.
The documentation does imply `MTL::DispatchTypeSerial` is the default

Reply:
I checked and `DispatchTypeSerial` is correct.